### DISCUSSION TITLE
What is the correct syntax to enable long captions (longer than a single line) in diagrams?

### DISCUSSION THREAD

**Discussion by Rockgecko** [quarto-dev/quarto-cli#1816] [Q&A] [answered]
The following is a minimal example of what I am trying to do:
```
---
title: "Test doc"
format:
  pdf:
    toc: false
    number-sections: true
    colorlinks: true
    documentclass: scrartcl
    papersize: A4
jupyter: python3
---

```{mermaid}
%%| label: fig-components
%%| fig-cap: "A long caption that I am using to demonstrate my problem
with long captions."
flowchart TB
  A(A) --> B(B)
  B --> C(C)
```

The caption (in reality much longer than this) wraps over several lines. In the VSCode file I get a warning `Key fig-cap has empty value but it must instead be a string`. 
I've tried using `\` separators at the end of lines, and using """ docstring-type quotes but these don't help.
Can't find an answer after searching the docs and github issues. 
Hope someone can help.
Thanks!

**Comment by jjallaire** [comment, marked as answer]
This requires a YAML multi-line string. Like this:

````markdown
```{mermaid}
%%| label: fig-components
%%| fig-cap: | 
%%|   A long caption that I am using to demonstrate my problem
%%|   with long captions
flowchart TB
  A(A) --> B(B)
  B --> C(C)
```
````

**Reply by Rockgecko** [reply]
Thanks, that's exactly what I needed.
